internal/engine/docker_pruner.go

```diff
@@ -0,0 +1,76 @@
+package engine
+
+import (
+	"context"
+	"time"
+
+	"github.com/windmilleng/tilt/internal/docker"
+	"github.com/windmilleng/tilt/internal/store"
+)
+
+// How often to prune Docker images while Tilt is running
+// TODO: configurable from Tiltfile for special cases
+const dockerPruneInterval = time.Hour * 12
+
+type DockerPruner struct {
+	dCli docker.Client
+
+	started            bool
+	disabledForTesting bool
+}
+
+func NewDockerPruner(dCli docker.Client) *DockerPruner {
+	return &DockerPruner{dCli: dCli}
+}
+
+func (dp *DockerPruner) DisableForTesting() {
+	dp.disabledForTesting = true
+}
+
+func (dp *DockerPruner) OnChange(ctx context.Context, st store.RStore) {
+	if dp.disabledForTesting {
+		return
+	}
+
+	if dp.started {
+		return
+	}
+
+	state := st.RLockState()
+	defer st.RUnlockState()
+
+	// wait until state has been kinda initialized
+	if !state.TiltStartTime.IsZero() && state.LastTiltfileError() == nil {
```

Comment:
rather than doing this on a background reaper process, i wonder if we should be more systematic about it? e.g., doing in when there are no pending builds, after N builds complete, etc

The case i'm most worried about is cases like Ellie + Nic where they're editing a file that produces a big binary, so want to reap every N edits, rather than tied to a specific time interval